pkg/controller/ps/controller.go

```diff
@@ -944,22 +936,24 @@ func (r *PerconaServerMySQLReconciler) reconcileMySQLRouter(ctx context.Context,
 }
 
 func (r *PerconaServerMySQLReconciler) cleanupOutdated(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
-	mysqlExposer := mysql.Exposer(*cr)
-	if err := r.cleanupOutdatedServices(ctx, cr, &mysqlExposer); err != nil {
-		return errors.Wrap(err, "cleanup MySQL services")
-	}
+	if !cr.Spec.Pause {
+		mysqlExposer := mysql.Exposer(*cr)
+		if err := r.cleanupOutdatedServices(ctx, cr, &mysqlExposer); err != nil {
+			return errors.Wrap(err, "cleanup MySQL services")
+		}
 
-	orcExposer := orchestrator.Exposer(*cr)
```

Comment:
shouldn't we add haproxy here to remove its services when necessary?

Reply:
Also, should we add the same for the `Router`?

Reply:
I will check it.

Reply:
@inelpandzic @egegunes we need to decide here how to do it better. For proxies we have separate method `cleanupProxies`
Now it looks a bit strange since we have methods for sts, services (type of resource) and proxies (function). So we should decide either we use methods by resource type (like cleanupOutdatedServices or cleanupOutdatedStatefulSets) or by function (like cleanupProxies)

Reply:
I would remove this `cleanupOutdatedServices` and `cleanupOutdatedStatefulSets` have everything by component in the `cleanupOutdate` method, like we have for proxies. So in the end we would have something like: 

```
cleanupOudated:
- cleanupMysql
- cleanupOrchestrator
- cleanupProxies (this one we already have)
```

This is way it is easier and cleaner it seems to me.

Reply:
I would rather do like this: 
- cleanupServices
- cleanupStatefulsets
- cleanupDeploments
Otherwise we will repeat the same code in all methods. I mean for both mysql and orc we should delete sts and services. 
@egegunes what do you think?

Reply:
Generally, we are treating these by component rather by resource type, thats why I don't this could be good.

Reply:
ok, np, if it our common behavior I will do like this)